framework/src/modules/chain/dpos/dpos.js

```diff
@@ -77,17 +79,99 @@ module.exports = class Dpos {
 
 	async onRoundFinish() {
 		const disposableDelegateList =
-			this.finalizedBlockRound - this.delegateListRoundOffset;
+			this.finalizedBlockRound -
+			this.delegateListRoundOffset -
+			this.delegateActiveRoundLimit;
 		await this.delegatesList.deleteDelegateListUntilRound(
 			disposableDelegateList,
 		);
 	}
 
+	async getActiveDelegateHeights(
+		numberOfRounds = 1,
+		{ tx, delegateListRoundOffset = this.delegateListRoundOffset },
+	) {
+		// TODO: Discuss reintroducing a caching mechanism to avoid fetching
+		// active delegate lists multiple times.
+		let delegateLists = await this.storage.entities.RoundDelegates.get(
```

Comment:
prefer no `let`.
Since type is different after map, it probably should have different name